calyx/src/passes/remove_if.rs

```diff
@@ -88,12 +88,18 @@ impl Visitor for RemoveIf {
                     }
                 }
 
-                let mut comps = vec![];
-                comps.append(&mut con.cond);
-                comps.append(&mut tbranch.comps.clone());
-                comps.append(&mut fbranch.comps.clone());
 
-                Ok(Action::Change(ast::Control::enable(comps)))
+                let mut comps_seq = vec![];
+                let mut comps1 = vec![];
+                let mut comps2 = vec![];
+                comps1.append(&mut con.cond);
+                comps2.append(&mut tbranch.comps.clone());
+                comps2.append(&mut fbranch.comps.clone());
```

Comment:
A different way of creating `comps2` is this:
```rust
                let branch_control: Vec<String> = tbranch.comps
                    .clone()
                    .into_iter()
                    .chain(fbranch.comps.clone().into_iter())
                    .collect();
```
this avoids creating an intermediate vector like `comps2` (as far as a I can tell).

Reply:
updated